.github/workflows/terraform_deploy.yaml

```diff
@@ -0,0 +1,34 @@
+name: Terraform deployment
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'environment name'
+        required: true
+
+jobs:
+  terraform_deployment:
+    environment: ${{ inputs.environment }}
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+    runs-on: ubuntu-latest
+    name: Run terraform
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Configure Terraform
+        uses: hashicorp/setup-terraform@v3
+
+      - name: Initialize Terraform
+        run: terraform init
+
+      - name: Run Terraform Plan
+        run: |
+          terraform plan -var="environment=${{ inputs.environment }}"
```

Comment:
Should we save the plan for use in the next step?

Reply:
is it necessary? it is all run in the same context.
does saving the plan speed up the 'terraform apply' afterward? I don't know enough :upside_down_face:

Reply:
I'm not sure about speeding things up, I think I know about as much as you do here :-)
Judging by https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS#automated-terraform-cli-workflow the saved plan is mostly intended for human review before actually applying it.

As an alternative question: will we need the plan stage once apply gets enabled?

Reply:
I don't think so, having the tfstate is all we need.
I'd like to merge this to try it out, we can improve things as we go :)